src/sock.c

```diff
@@ -95,8 +95,13 @@ int opensock (const char *host, int port, const char *bind_to)
 
         n = getaddrinfo (host, portstr, &hints, &res);
         if (n != 0) {
+                const char *error;
```

Comment:
hmm, i think we should pack this into a function rather than duplicating the block further down

Reply:
With a function like this?

```
const char * get_gai_error(int n) {
                if (n == EAI_SYSTEM)
                        return strerror (errno);
                else
                        return gai_strerror (n);
}
```

Reply:
right! but it should be static